var/spack/repos/builtin/packages/pythia6/package.py

```diff
@@ -33,11 +33,13 @@ class Pythia6(CMakePackage):
     """
 
     homepage = 'https://pythiasix.hepforge.org/'
-    url = 'http://www.hepforge.org/archive/pythiasix/pythia-6.4.28.tgz'
+    url = 'https://pythia.org/download/pythia6/pythia6428-split.tgz'
```

Comment:
Is this definition of `url` even necessary if we're not using it anywhere?

Reply:
It's used to search for new versions of the package.

Reply:
We got an error because the old url did not exist. Once the url was updated, spack proceeded to use its default source cache for this package. Perhaps this was because the checksum did not change. Removing the default mirror was the only was to have spack download from the new url.

Reply:
So it seems that spack always checks the url.

Reply:
> We got an error because the old url did not exist.

Can you open an issue with the exact error message and steps to reproduce? This should not be the case.

Reply:
This error occurs when no spack-public mirror is configured.